lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -705,3 +716,263 @@ def modify_learner_enrollment_status(self, enrollment_request, program_enrollmen
+            * micromasters_title: title of the MicroMasters program that the course run is a part of;
+                if absent, the course run is not a part of a MicroMasters program
+            * certificate_download_url: url to download a certificate, if available;
+                if absent, certificate is not downloadable
+
+    **Example**
+
+        {
+            "course_runs": [
+                {
+                    "course_run_id": "edX+AnimalsX+Aardvarks",
+                    "display_name": "Astonishing Aardvarks",
+                    "course_run_url": "https://courses.edx.org/courses/course-v1:edX+AnimalsX+Aardvarks/course/",
+                    "start_date": "2017-02-05T05:00:00Z",
+                    "end_date": "2018-02-05T05:00:00Z",
+                    "course_status": "completed"
+                    "emails_enabled": true,
+                    "due_dates": [
+                        {
+                            "name": "Introduction: What even is an aardvark?",
+                            "url": "https://courses.edx.org/courses/course-v1:edX+AnimalsX+Aardvarks/jump_to/block-v1:edX+AnimalsX+Aardvarks+type@chapter+block@1414ffd5143b4b508f739b563ab468b7",
+                            "date": "2017-05-01T05:00:00Z"
+                        },
+                        {
+                            "name": "Quiz: Aardvark or Anteater?",
+                            "url": "https://courses.edx.org/courses/course-v1:edX+AnimalsX+Aardvarks/jump_to/block-v1:edX+AnimalsX+Aardvarks+type@sequential+block@edx_introduction",
+                            "date": "2017-03-05T00:00:00Z"
+                        }
+                    ],
+                    "micromasters_title": "Animals",
+                    "certificate_download_url": "https://courses.edx.org/certificates/123"
+                },
+                {
+                    "course_run_id": "edX+AnimalsX+Baboons",
+                    "display_name": "Breathtaking Baboons",
+                    "course_run_url": "https://courses.edx.org/courses/course-v1:edX+AnimalsX+Baboons/course/",
+                    "start_date": "2018-02-05T05:00:00Z",
+                    "end_date": null,
+                    "course_status": "in-progress"
+                    "emails_enabled": false,
+                    "due_dates": [],
+                    "micromasters_title": "Animals",
+                    "certificate_download_url": "https://courses.edx.org/certificates/123",
+                    "resume_course_run_url": "https://courses.edx.org/courses/course-v1:edX+AnimalsX+Baboons/jump_to/block-v1:edX+AnimalsX+Baboons+type@sequential+block@edx_introduction"
+                }
+            ]
+        }
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS,)
```

Comment:
I'm not sure about this. I need to dig in a little to see whether this is actually necessary. I'm having trouble with this in testing.